src/analogWrite.cpp

```diff
@@ -34,6 +30,12 @@ void analogWrite(uint8_t APin, uint16_t AValue) {
 		}
 		digitalWrite(APin, 1);
 	} else
+	{
+		if(!chan->attached()){
+			chan->attachPin(APin,1000, 8); // This adds the PWM instance to the factory list
```

Comment:
this is unreachable because of the same check above

Reply:
Please have a second look at it: I deleted the check above in my proposal, because I don't want the check to be executed in case of value 0 or 255, as it tries to allocated a PWM channel.